google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java

```diff
@@ -1612,4 +1615,70 @@ public void testLargeRequest() throws IOException, InterruptedException, Executi
       assertEquals("50", queryIter.next().get(0).getStringValue());
     }
   }
+
+  @Test
+  public void testTimeoutException() throws IOException, InterruptedException, ExecutionException {
+    try {
+      JsonStreamWriter.setMaxRequestCallbackWaitTime(
+          Duration.ofMillis(10)); // induce timeout exception
```

Comment:
I am not entirely sure if tests will be executed in parallel, they usually will. If this is a static var, it could be problematic. I think it is safer to unit test this? In unit test, we can delay the callback for as long as we want, is the unit test not enough to hit the problem?

Reply:
Or we could introduce a local timeout var on the object, maybe call it SetLocalMaxRequestCallbackWaitTime or something. Maybe this value shouldn't be global to start with... But if we change the behavior now, it would be a breaking change.